In [11]:
import pandas as pd

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#Healthcare Dataset Stroke Data

### Source of data:


https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

### Description:


According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

### Target

Target is "Stroke" as function of contributing factors that share a commonality with people who have strokes

### What does one row represent?

The row represents a patient

In [4]:
import pandas as pd
filename = '/content/drive/MyDrive/CodingDojo/Beginning Data Science/Projects/healthcare-dataset-stroke-data.csv'
df = pd.read_csv(filename)
df.head(1)

id gender   age  hypertension  heart_disease ever_married work_type  \
0  9046   Male  67.0             0              1          Yes   Private   

  Residence_type  avg_glucose_level   bmi   smoking_status  stroke  
0          Urban             228.69  36.6  formerly smoked       1

### Is this a classification or regression problem?

This is a classification problem

###  How many features does the data have?

In [5]:
df.shape

(5110, 12)

This dataset has 12 features

### How many rows are in this dataset?


5110 rows

### What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

In [6]:
#Investigating Missing Values
df[df.isna().any(axis=1)].shape

(201, 12)

In [7]:
df.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

Moderate risk in resolving missing values on what is a key indicator for risk.  Error in inputation could ruin investigation.

#Heart Failure Prediction Dataset

In [12]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


### Source of data:

https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

### Description:

Cardiovascular diseases (CVDs) are the number 1 cause of death globally, taking an estimated 17.9 million lives each year, which accounts for 31% of all deaths worldwide. Four out of 5CVD deaths are due to heart attacks and strokes, and one-third of these deaths occur prematurely in people under 70 years of age. Heart failure is a common event caused by CVDs and this dataset contains 11 features that can be used to predict a possible heart disease.

People with cardiovascular disease or who are at high cardiovascular risk (due to the presence of one or more risk factors such as hypertension, diabetes, hyperlipidaemia or already established disease) need early detection and management wherein a machine learning model can be of great help.

### Target

The target for this investigation is heart disease as a function of predicting heart failure

### What does one row represent?

In [13]:
import pandas as pd
filename = '/content/drive/MyDrive/CodingDojo/Beginning Data Science/Projects/heart.csv'
df = pd.read_csv(filename)
df.head(1)

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0

The row is an investigation of individuals presenting an array of condtions that fall in the domanin of heart disease

### Is this a classification or regression problem?

This is a classification problem since it is investigating those with and without heart disease

### How many features does the data have?

In [16]:
df.shape

(918, 12)

There are 12 features

### How many rows are in this dataset?

There are 918 rows

### What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

In [17]:
#Investigating Missing Values
df[df.isna().any(axis=1)].shape

(0, 12)

In [18]:
df.isna().sum()

Age               0
Sex               0
ChestPainType     0
RestingBP         0
Cholesterol       0
FastingBS         0
RestingECG        0
MaxHR             0
ExerciseAngina    0
Oldpeak           0
ST_Slope          0
HeartDisease      0
dtype: int64

In [19]:
df.duplicated().sum()

0

There doesnt appear to be major issue with this data frame